### PR TITLE
[app] Add layout components

### PR DESCRIPTION
**Pull request by zerefwayne**
This PR adds layout components to the app.

- Header
- Sidenav 

image

### CODE REVIEW

File: diagnostics-app/src/app/app.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }).compileComponents();
+  });
+
+  it('should create the app', () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+    expect(app).toBeTruthy();
+  });
+
+  // it(`should have as title 'diagnostics-app'`, () => {
```
Comment (Lamzin) [inline diagnostics-app/src/app/app.component.spec.ts:29]: Please remove commented code.

File: diagnostics-app/.gitignore [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# See http://help.github.com/ignore-files/ for more about ignoring files.
```
Comment (Lamzin) [inline diagnostics-app/.gitignore:1]: Do we need a second gitignore file? Can't we have only one per repo?

File: diagnostics-app/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "scripts": {
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "watch": "ng build --watch --configuration development",
+    "test": "ng test"
+  },
+  "private": true,
+  "dependencies": {
+    "@angular/animations": "~12.0.2",
```
Comment (Lamzin) [inline diagnostics-app/package.json:13]: Does package.json support comments? I'd like to add a brief comment to each dependency explaining what is it. WDYT?

File: diagnostics-app/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  "version": "0.0.0",
+  "scripts": {
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "watch": "ng build --watch --configuration development",
+    "test": "ng test"
+  },
+  "private": true,
+  "dependencies": {
```
Comment (Lamzin) [inline diagnostics-app/package.json:12]: Can you please go through each dependency and double check whether we need all of them at this point?

File: diagnostics-app/src/app/header/header.component.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  darkThemeEnabled = true;
+  constructor() {}
+
+  ngOnInit(): void {
+    this.themeChange.emit(this.darkThemeEnabled);
+  }
+
+  onToggleTheme() {
+    // TODO Enable toggle after deciding light theme
```
Comment (Lamzin) [inline diagnostics-app/src/app/header/header.component.ts:25]: same: please remove commented code

File: diagnostics-app/src/app/sidenav/sidenav-list/sidenav-list.component.css [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+h2 {
+  margin-top: .5rem;
+  padding-left: 2rem;
+  margin-bottom: .5rem;
+  color: #aaa;
+}
```
Comment (Lamzin) [inline diagnostics-app/src/app/sidenav/sidenav-list/sidenav-list.component.css:12]: nit: please add \n at the end

File: diagnostics-app/src/app/sidenav/sidenav.component.css [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+ */
+
+ul {
+  margin: 0;
+  padding: 0;
+}
+
+.app-sidenav {
+  padding-top: 1rem;
+}
```
Comment (Lamzin) [inline diagnostics-app/src/app/sidenav/sidenav.component.css:14]: nit: same

File: diagnostics-app/src/polyfills.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
+/**
```
Comment (Lamzin) [inline diagnostics-app/src/polyfills.ts:1]: Do we actually need polyfills?

**Review by Lamzin — COMMENTED** [review]
Hi,

I didn't see that this PR is a draft, so started with this one, but probably most comments related to the first PR.

`diagnostics-app/package-lock.json` looks automatically generated, should we really have it in the git tree?

**Comment by MahmoudAGawad** [comment]
Looks like this PR include all changes from the first PR. Can you fix this?

**Comment by zerefwayne** [comment]
> Looks like this PR include all changes from the first PR. Can you fix this?

@MahmoudAGawad @Lamzin I've fixed most of the comments mentioned here in PR #1 Since this PR contains so many changes, I'll split this into multiple PRs after the first one gets merged.

**Comment by zerefwayne** [comment]
After discussion, it was decided to scrap this Pull Request as a better alternative was found.

**Pull request closed without merging**